rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -323,20 +333,50 @@ def _label_id_index_mapping(
     def _invert_mapping(mapping: Dict) -> Dict:
         return {value: key for key, value in mapping.items()}
 
-    def _tag_id_index_mapping(self, training_data: TrainingData) -> Dict[Text, int]:
-        """Create tag_id dictionary"""
+    def _tag_id_index_mappings(
+        self, training_data: TrainingData
+    ) -> Dict[Text, Dict[Text, int]]:
+        """Create tag_id dictionaries for all CRFs."""
 
-        if self.component_config[BILOU_FLAG]:
-            return bilou_utils.build_tag_id_dict(training_data)
+        tag_id_index_mapping_dict = {}
 
-        distinct_tag_ids = {
-            e["entity"]
-            for example in training_data.entity_examples
-            for e in example.get(ENTITIES)
-        } - {None}
+        for attribute_key in CRF_ORDER:
+            if (
+                attribute_key == ENTITY_ATTRIBUTE_TYPE
```

Comment:
should `BILOU` be added for all tags?

Reply:
Not sure, I can quickly run an experiment, but my assumption is that it does not change much. We also forward the logits from the first CRF, that includes BILOU, to the second level CRFs.

Reply:
Seems to be slightly better when we apply BILOU to all tags. Will update the code.